### ISSUE TITLE
write_canonical/1: Variables written differently

### ISSUE THREAD

**Issue by UWN** [infradig/trealla#399]
```
Trealla Prolog (c) Infradig 2020-2021, v1.19.27
?-  write_canonical(_+_). 
+(_,_)   true. % unexpected
?- write_canonical(B+B). 
+(A,A)   true. % unexpected
```

They must start with `_` and be different for different variables.

[#226](http://www.complang.tuwien.ac.at/ulrich/iso-prolog/conformity_testing#226) etc.

[7.10.5 a2](http://www.complang.tuwien.ac.at/ulrich/iso-prolog/draft-7.10.5#a2).

**Comment by UWN** [comment]
Nothing is fixed, currently:

```
ulrich@p0:/opt/gupu/trealla$ git pull&&make OPT=-std=c99 NOSSL=1&&kill %1;date;git describe;time ./tpl -f
Already up-to-date.
cc -Isrc -I/usr/local/include -DVERSION='"v1.20.4"' -O3 -Wall -Wextra -D_GNU_SOURCE -Wno-deprecated-declarations -funsigned-char -std=c99   -c -o src/module.o src/module.c
src/module.c: In function ‘asserta_to_db’:
src/module.c:902:2: error: a label can only be part of a statement and a declaration is not a statement
  db_entry *dbe = assert_begin(m, nbr_vars, p1, consulting);
  ^
src/module.c: In function ‘assertz_to_db’:
src/module.c:950:2: error: a label can only be part of a statement and a declaration is not a statement
  db_entry *dbe = assert_begin(m, nbr_vars, p1, consulting);
  ^
make: *** [src/module.o] Error 1
Mon Jan 17 09:14:04 CET 2022
v1.20.4
Trealla Prolog (c) Infradig 2020-2021, v1.19.27
?- 
````

**Comment by UWN** [comment]
Fixed in v1.20.0 (after some git reset --hard HEAD^^^^^)
